packages/markers/src/browser/problem/problem-contribution.ts

```diff
@@ -119,6 +154,38 @@ export class ProblemContribution extends AbstractViewContribution<ProblemWidget>
         }
     }
 
+    protected handleCopy(content: string): void {
+        const cp = document.createElement('textarea');
+        cp.value = content;
+        document.body.appendChild(cp);
+        cp.select();
+        document.execCommand('copy');
+        document.body.removeChild(cp);
+    }
+
+    protected problemCopy(selection: ProblemSelection): void {
+        const marker = selection.marker as ProblemMarker;
+        const serializedProblem = JSON.stringify({
+            resource: marker.uri,
+            owner: marker.uri,
+            code: marker.data.code,
+            severity: marker.data.severity,
+            message: marker.data.message,
+            source: marker.data.source,
+            startLineNumber: marker.data.range.start.line,
+            startColumn: marker.data.range.start.character,
+            endLineNumber: marker.data.range.end.line,
+            endColumn: marker.data.range.end.character
+        }, undefined, '\t');
+
+        this.handleCopy(serializedProblem);
+    }
+
+    protected problemCopyMessage(selection: ProblemSelection): void {
```

Comment:
Same as [above](https://github.com/theia-ide/theia/pull/5162/files#r287543094), the name `copyMessage(...` would be sufficient.